doc/modules/linear_model.rst

```diff
@@ -1051,6 +1057,62 @@ considering only a random subset of all possible combinations.
 
     .. [#f2] T. Kärkkäinen and S. Äyrämö: `On Computation of Spatial Median for Robust Data Mining. <http://users.jyu.fi/~samiayr/pdf/ayramo_eurogen05.pdf>`_
 
+.. _huber_regression:
+
+Huber Regression
+----------------
+
+The :class:`HuberRegressor` is an extension of :class:`Ridge` except that
+the loss above a certain threshold is linear. It differs from :class:`TheilSenRegressor`
+and :class:`RANSACRegressor` because it does not ignore the effect of the outliers
+but gives a lesser weight to them.
+
+.. figure:: ../auto_examples/linear_model/images/plot_huber_vs_ridge_001.png
+   :target: ../auto_examples/linear_model/plot_huber_vs_ridge.html
+   :align: center
+   :scale: 50%
+
+The loss function that :class:`HuberRegressor` minimizes is given by
+
+.. math::
+
+   \underset{w, \sigma}{min\,} {n_samples*\sigma + H_m(\frac{X w - y}{\sigma})*\sigma + \alpha {||w||_2}^2}
+
+where
+
+.. math:: H_m(z)
+
+.. math:: z^2 if |z| < \epsilon
+
+.. math:: 2*M*|z| - M^2 if |z| >= \epsilon
+
+It is advised to set the parameter `epsilon` to 1.35 to acheive 95% statistical efficiency.
+
+Notes
+-----
+The :class:`HuberRegressor` differs from using :class:`SGDRegressor` with loss set to `huber`
+in the following ways.
+
+- :class:`HuberRegressor` is scaling invariant. Once `epsilon` is set, scaling X and y
+  down or up by different values would produce the same robustness to outliers as before.
+  as compared to :class:`SGDRegessor` where `epsilon` has to be set again when X and y are
+  scaled.
+
+- :class:`HuberRegressor` should be more efficient to use on data with small number of
+  samples while :class:`SGDRegressor` needs a number of passes on the training data to
```

Comment:
Yes ! When we have small number of samples, we need a large number of passes to acheive the same robustness as the HuberRegressor.